Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -895,6 +896,37 @@ void RenderLayer::collectLayers(std::unique_ptr<Vector<RenderLayer*>>& positiveZ
     }
 }
 
+void RenderLayer::setNeedsPositionUpdate()
+{
+    m_layerPositionDirtyBits.add(LayerPositionUpdates::NeedsPositionUpdate);
+    for (auto* layer = parent(); layer; layer = layer->parent()) {
+        if (layer->m_layerPositionDirtyBits.contains(LayerPositionUpdates::DescendantNeedsPositionUpdate))
+            break;
+        layer->m_layerPositionDirtyBits.add(LayerPositionUpdates::DescendantNeedsPositionUpdate);
+    }
+}
+
+bool RenderLayer::needsPositionUpdate() const
+{
+    if (m_layerPositionDirtyBits.containsAny({ LayerPositionUpdates::NeedsPositionUpdate, LayerPositionUpdates::DescendantNeedsPositionUpdate }))
+        return true;
+    if (parent() && parent()->m_layerPositionDirtyBits.contains(LayerPositionUpdates::AllChildrenNeedPositionUpdate))
+        return true;
+    return false;
```

Comment:
nit. why not just `return parent() && parent()->m_layerPositionDirtyBits.contains(LayerPositionUpdates::AllChildrenNeedPositionUpdate);`